rust/connlib/tunnel/src/tests.rs

```diff
@@ -96,6 +104,10 @@ enum Transition {
     SendICMPPacketToIp4Resource { r_idx: sample::Index },
     /// Send a ICMP packet to an IPv6 resource.
     SendICMPPacketToIp6Resource { r_idx: sample::Index },
+    /// The system's DNS servers changed.
+    UpdateSystemDnsServers { servers: Vec<IpAddr> },
+    /// The upstream DNS servers changed.
+    UpdateUpstreamDnsServers { servers: Vec<DnsServer> },
```

Comment:
What if we use an interface config here?

Reply:
We can extend that too but AFAIK, the portal never re-assigns us new IPs, right? So I didn't capture that yet. I think it may also make sense to model that as a separate transition because it is somewhat orthogonal. The fact that the portal hands up both, the IPs and the upstream DNS servers in one message is kind of unrelated IMO?

Plus, IMO `InterfaceConfig` is a type that I consider "internal" to connlib. Once we tackle https://github.com/firezone/firezone/issues/5099, this will become a lot more obvious because all the "message" types will be implementation details of `Tunnel`.